### DISCUSSION TITLE
Local (web) storage affected by "Experimental third-party storage partitioning"

### DISCUSSION THREAD

**Discussion by doubletuna** [privacysandbox/privacy-sandbox-dev-support#157] [Replacing cross-site tracking] [answered]
Hello,

In short: I'd like to share local storage between two iframes of the same origin and not under the same parent.
(share = being able to fully utilize the local storage, get/set)

using Chrome 117.x, site x.com opens my website as an iframe, for auth, which in turn saves a token to local storage. this iframe is not accessible after this process is complete, site x.com closes the iframe.

At a later stage, site x.com opens my site again, only this time, I'm expecting the local storage to contain a token so I can seamlessly login.

Up until "third-party storage partitioning" started rolling out, everything went smooth and now, with "third-party storage partitioning" set as Enabled (or even in some cases, randomly, at Default) it doesn't work anymore. My site cannot access the local storage that the original iframe had set.

Has anyone encountered a similar problem? Is there a working example of a solution for this issue (using requestStorageAccess for example)?

Looking for a solution that will preferably resolve this issue without user interaction (if possible)

Cheers

**Comment by miketaylr** [comment, marked as answer]
Hi there, yes, this is expected behavior given the feature (and the same way that Firefox and Safari behave). There is a deprecation trial available for approximately one year: https://developer.chrome.com/blog/storage-partitioning-deprecation-trial/

And we are exploring an option to allow `requestStorageAccess` to be used to regain a handle to unpartitioned storage. See https://groups.google.com/a/chromium.org/g/blink-dev/c/inRN8tI49O0/m/SJNXk6ISAgAJ
